src/main/webapp/intrigue-api/metacards/metacards.js

```diff
@@ -238,10 +238,41 @@ const createThumbnailUrl = result => {
   return 'data:image/jpeg;base64,' + thumbnail
 }
 
+const getResultFormAttributes = async (resultFormID, catalog) => {
+  if (!resultFormID || resultFormID === 'All Fields') return null
+  try {
+    const filterTree = {
+      type: 'AND',
+      filters: [
+        { type: '=', property: 'id', value: resultFormID },
+        { type: 'LIKE', property: 'metacard-tags', value: '%' },
+      ],
+    }
+    const q = { filterTree }
+    const json = await catalog.query(processQuery(q))
+    return json.results[0].metacard.attributes['ui.attribute-group']
+  } catch (e) {
+    return null
+  }
+}
+
+const filterResultFormAttributes = (attributes, resultFormAttributes) => {
+  if (!resultFormAttributes) return attributes
+  return Object.keys(attributes).reduce((acc, current) => {
+    if (resultFormAttributes.includes(current) || current === 'id') {
+      return { ...acc, [current]: attributes[current] }
+    }
+    return acc
+  }, {})
+}
+
 const metacards = async (parent, args, { catalog, toGraphqlName, fetch }) => {
   const q = { ...args.settings, filterTree: args.filterTree }
   const json = await catalog.query(processQuery(q))
-
+  const resultFormAttributes = await getResultFormAttributes(
```

Comment:
should await both of these async calls at the same time instead of awaiting them one by one